syntax = "proto2";


message logline {
        enum loglevel {
             FORCED = 0;
             FATAL = 1;
             ERROR = 10;
             WARN = 20;
             INFO = 30;
             DEBUG = 100;
             DEBUG_NOISY = 200;
            };
        
        optional loglevel level = 1 [default = DEBUG];
        optional string content = 2;
};

message argali_packet {

  enum packet_type {
    LOGLINE_PKT = 1;
  };

  required packet_type payload_type = 1;
  oneof msg {
    logline logline = 2;
  }
};